packages/flutter_tools/lib/src/commands/format.dart

```diff
@@ -40,26 +41,36 @@ class FormatCommand extends FlutterCommand {
       'format',
     ];
     final List<String> rest = argResults?.rest ?? <String>[];
-    if (rest.isEmpty) {
-      globals.printError(
-        'No files specified to be formatted.'
-      );
+    if (rest.isEmpty || onlyHasOptions(rest)) {
+      globals.printError('No files specified to be formatted.');
       command.add('-h');
     } else {
       command.addAll(<String>[
         for (String arg in rest)
-          if (arg == '--dry-run' || arg == '-n')
-            '--output=none'
-          else
-            arg,
+          if (arg == '--dry-run' || arg == '-n') '--output=none' else arg,
       ]);
     }
 
-    final int result = await globals.processUtils.stream(command);
-    if (result != 0) {
-      throwToolExit('Formatting failed: $result', exitCode: result);
+    final RunResult result = globals.processUtils.runSync(command);
```

Comment:
we want stream here, right?

Reply:
I guess this is a nit. The main reason to use `stream` over `runSync` would be because we are only interested in the exit code and would just avoid using an extra else in this case, but I have seen `runSync`/`run' used more overall on tool thats why I changed it to that.